preheaders/billing.php

```diff
@@ -276,7 +276,11 @@
 			
 			do_action( 'pmpro_after_update_billing', $current_user->ID, $morder );
         } else {
-            $pmpro_msg = $morder->error;
+			// Track as spam activity to prevent too many CC attempts.
```

Comment:
Can we add a hook here instead perhaps, and have `pmpro_track_spam_activity` trigger on that hook?

Thinking about a hook here because it may be useful to allow people (or us) to implement other solutions to this problem via add ons.

Reply:
I like it. We can add a pmpro_update_billing_failed hook here, and a similar pmpro_checkout_process_failed on the checkout page then hook into them.

The pmpro_track_spam_activity function can be called with no params so it's easy to hook up.